src/app/shared/filter/date/date-range.component.ts

```diff
@@ -0,0 +1,30 @@
+import { Component, Input, Output, EventEmitter } from '@angular/core';
+import { FilterModel, DateRangeModel } from '../../../ngrx/model';
+import { getBeginEndDateFromWhen, getWhenForRange, getRange } from '../../util/date.util';
+
+@Component({
+  selector: 'metrics-date-range',
+  template: `
+    <metrics-standard-date-range *ngIf="filter" [interval]="filter.interval" [when]="filter.when"
+                                 (whenChange)="onWhenChange($event)"></metrics-standard-date-range>
+    <metrics-custom-date-range *ngIf="filter" [interval]="filter.interval"
+                               [beginDate]="filter.beginDate" [endDate]="filter.endDate"
+                               (dateRangeChange)="onDateRangeChange($event)"></metrics-custom-date-range>
+  `
+})
+export class DateRangeComponent {
+  @Input() filter: FilterModel;
+  @Output() dateRangeChange = new EventEmitter<DateRangeModel>();
+
+  onDateRangeChange(dateRange: DateRangeModel) {
+    const when = getWhenForRange(dateRange);
+    const range = getRange(when);
+    this.dateRangeChange.emit({...dateRange, when, range});
```

Comment:
Feels slightly odd to have two different "levels" of event emitters both called `dateRangeChange`? Will the dateRangeChange emitted by the custom date range propagate up to the metrics-filter component as well?

Reply:
only this one gets up to the metrics-filter, but I can rename the lower one to customDateRangeChange if that makes it less confusing

Reply:
i think that could help!